bn-apps/memberships-management/membership-service/src/main/kotlin/com/r3/businessnetworks/membership/flows/bno/RequestMembershipFlowResponder.kt

```diff
@@ -27,23 +28,23 @@ import java.sql.SQLException
  * TODO: remove MembershipAutoAcceptor in favour of flow overrides when Corda 4 is released
  */
 @InitiatedBy(RequestMembershipFlow::class)
-class RequestMembershipFlowResponder(val session : FlowSession) : BusinessNetworkOperatorFlowLogic<Unit>() {
+open class RequestMembershipFlowResponder(val session : FlowSession) : BusinessNetworkOperatorFlowLogic<Unit>() {
 
     @Suspendable
     override fun call() {
         // checking that there is no existing membership state
         val counterparty = session.counterparty
         val configuration = serviceHub.cordaService(BNOConfigurationService::class.java)
         val databaseService = serviceHub.cordaService(DatabaseService::class.java)
-        val counterpartMembership = databaseService.getMembership(counterparty, ourIdentity, configuration.membershipContractName())
+        val counterpartMembership = databaseService.getMembership(counterparty, ourIdentity)
         if (counterpartMembership != null) {
             throw FlowException("Membership already exists")
         }
 
         // creating a pending request to make sure that no multiple on-boarding request can be in-flight in the same time
         try {
             databaseService.createPendingMembershipRequest(session.counterparty)
-        } catch (e : SQLException) {
+        } catch (e : PersistenceException) {
```

Comment:
Should this also catch `TransactionRequiredException`?

Reply:
Changed in the favour of `PersistenceException`